plugins/modules/identity/sap_company.py

```diff
@@ -0,0 +1,331 @@
+      conn_username: 'DDIC'
+      conn_password: 'HECtna2021#'
+      host: 100.0.201.20
+      sysnr: '01'
+      client: '000'
+      state: absent
+      company_id: "Comp_ID"
+      name: "Test_comp"
+      name_2: "LTD"
+      country: "DE"
+      time_zone: "UTC"
+      city: "City"
+      post_code: "12345"
+      street: "test_street"
+      street_no: "1"
+      e_mail: "test@test.de"
+
+'''
+
+RETURN = r'''
+# These are examples of possible return values, and in general should use other names for return values.
+msg:
+  description: A small execution description.
+  type: str
+  returned: always
+  sample: 'Company address COMP_ID created'
+out:
+  description: A complete description of the executed tasks. If this is available.
+  type: list
+  elements: dict
+  returned: always
+  sample: '{
+        "RETURN": [
+                {
+                "FIELD": "",
+                "ID": "01",
+                "LOG_MSG_NO": "000000",
+                "LOG_NO": "",
+                "MESSAGE": "Company address COMP_ID created",
+                "MESSAGE_V1": "COMP_ID",
+                "MESSAGE_V2": "",
+                "MESSAGE_V3": "",
+                "MESSAGE_V4": "",
+                "NUMBER": "078",
+                "PARAMETER": "",
+                "ROW": 0,
+                "SYSTEM": "",
+                "TYPE": "S"
+                }
+            ]
+        }
+    }'
+'''
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+import traceback
+try:
+    from pyrfc import Connection
+except ImportError:
+    HAS_PYRFC_LIBRARY = False
+    ANOTHER_LIBRARY_IMPORT_ERROR = traceback.format_exc()
+else:
+    HAS_PYRFC_LIBRARY = True
+
+
+def call_rfc_method(connection, method_name, kwargs):
+    # PyRFC call function
+    return connection.call(method_name, **kwargs)
+
+
+def build_company_params(name, name_2, country, time_zone, city, post_code, street, street_no, e_mail):
+    # Creates RFC parameters for creating organizations
+    # define dicts in batch
+    params = dict()
+    # define company name
+    params['NAME'] = name
+    params['NAME_2'] = name_2
+    # define location
+    params['COUNTRY'] = country
+    params['TIME_ZONE'] = time_zone
+    params['CITY'] = city
+    params['POSTL_COD1'] = post_code
+    params['STREET'] = street
+    params['STREET_NO'] = street_no
+    # define communication
+    params['E_MAIL'] = e_mail
+    # return dict
+    return params
+
+
+def return_analysis(raw):
+    change = False
+    failed = False
+    msg = raw['RETURN'][0]['MESSAGE']
+    for state in raw['RETURN']:
+        if state['TYPE'] == "E":
+            if state['NUMBER'] == '081':
+                change = False
+            else:
+                failed = True
+        if state['TYPE'] == "S":
+            if state['NUMBER'] != '079':
+                change = True
+            else:
+                msg = "No changes where made."
+    return [{"change": change}, {"failed": failed}, {"msg": msg}]
```

Comment:
not defined for any other RETURN state than 'E' or  'S'. Maybe explicitly fail for other state than those two?
e.g.
```
if state['TYPE'] == 'E':
...
elif state['TYPE'] == 'S':
...
else raise TypeError 
```